### PR TITLE
Major feature: match()

### PR DESCRIPTION
**Pull request by andywer**
This is quite a big pull request (sorry!), but it comes with a bunch of neat improvements, I think:

* Introduces `match()` to get rid of inconsistent `exclude`/`include` options in blocks
* API of most blocks didn't change or just dropped `exclude`/`include` options
* API of `@webpack-blocks/assets` (`css()`, `file()` & `url()` block) changed noticeably
* Docs have been updated

Needed to touch all blocks for this (😱), but I used this opportunity to improve the block tests a bit and add a few new tests. Also cleaned up some nasty code bits.

Please have a look, do a quick check of the changed code and read the updated docs if you can find some time. I would like to publish this as `1.0.0-beta`.

Shout if you disagree with something, I was out on a limb here 😉

### CODE REVIEW

**Comment by andywer** [comment]
I know I touched a lot of stuff, but I hope you will find some time to quick-review the PR nevertheless.

If you are in a hurry, just check out the changed markdown files to see what changes for the user instead of reviewing the under-the-hood changes 😉

@jvanbruegge @sapegin @aaronjensen @eXon @diegohaz @bkonkle

File: packages/postcss/__tests__/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        {
+          loader: 'postcss-loader',
+          options: {}
+        }
+      ]
+    }
+  ])
+})
+
+test('Postcss works with css() & match()', t => {
```
Comment (jvanbruegge) [inline packages/postcss/__tests__/integration.test.js]: I would add a sass & match test case

Comment (andywer) [inline packages/postcss/__tests__/integration.test.js]: Sure 👍

Comment (andywer) [inline packages/postcss/__tests__/integration.test.js]: Done. But I added it to the global integration test (`packages/webpack/__tests__/integration.test.js`).

**Comment by andywer** [comment]
Last chance to review this PR. Going to merge it today or tomorrow.

Will publish `v1.0.0-beta` thereafter 🎉

**Comment by sapegin** [comment]
I think we should get rid of `context.fileType` entirely — I don’t see any benefits of having it now. We could just use globs everywhere and replace it with something like `context.testFor('*.css')`.

**Comment by andywer** [comment]
Fair enough. I am not sure we even need a `context.testFor()`. Could just use `test: context.match.test || /\.css$/`.

In this case we would just change our loaders and deprecate `context.fileType`, yielding a warning if you still use it.

**Comment by andywer** [comment]
What do you mean by "if it’s only for custom loaders"?

**Comment by sapegin** [comment]
Blocks, not loaders ;-)

**Comment by bkonkle** [comment]
>I think we should get rid of context.fileType entirely

I agree - there's not really a need for it any more, and it unnecessarily complicates things in my opinion. I like the `match()` api much better. 👍

**Comment by andywer** [comment]
I pushed a commit to replace all usages of `context.fileType` in webpack-blocks and mention in the docs and code that it is deprecated.

I'd say this can now be published as 1.0 beta. We can introduce a warning message in the next version and remove it completely in the version thereafter.

**Comment by bkonkle** [comment]
Awesome! 🎉

**Comment by andywer** [comment]
Published as `v1.0.0-beta`

**Pull request merged**